industrial_ci/ci_main.sh

```diff
@@ -0,0 +1,335 @@
+#!/bin/bash
```

Comment:
Yest it's a rename and move. 

Removed variables:

- `CI_PARENT_DIR`: no more needed.
- CI_SOURCE_PATH --> `TARGET_REPO_PATH`: Better description of the purpose.
- DOWNSTREAM_REPO_NAME --> `TARGET_REPO_NAME`: Better description of the purpose.

Change made is due to the directory change (see diff below. Apologies for inconvenient old style):
```
diff --git a/old/industrial_ci/travis.sh b/new/industrial_ci/industrial_ci/ci_main.sh
index 3b90696..c38f605 100755
--- a/./travis.sh
+++ b/home/n130s/link/industrial_ci/industrial_ci/ci_main.sh
@@ -1,3 +1,4 @@
+s
 #!/bin/bash
 
 # Software License Agreement (BSD License)
@@ -34,18 +35,16 @@
 
 ## This is a "common" script that can be run on travis CI at a downstream github repository.
 ## See ./README.rst for the detailed usage.
-##
-## Variables that are not meant to be exposed externally from this script may be lead by underscore.
 
 set -e
 set -x
 
 # Define some env vars that need to come earlier than util.sh
-export CI_SOURCE_PATH=$(pwd)
-export CI_PARENT_DIR=.ci_config  # This is the folder name that is used in downstream repositories in order to point to this repo.
+export ICI_PKG_PATH=$(pwd)  # The path on CI service (e.g. Travis CI) of industrial_ci top dir.
+export CI_MAIN_PKG=industrial_ci
 export HIT_ENDOFSCRIPT=false
 
-source ${CI_SOURCE_PATH}/$CI_PARENT_DIR/util.sh
+source ${ICI_PKG_PATH}/util.sh
 
 trap error ERR
 trap success SIGTERM  # So that this script won't terminate without verifying that all necessary steps are done.
@@ -54,7 +53,7 @@ trap success SIGTERM  # So that this script won't terminate without verifying th
 # The Dockerfile in this repository defines a Ubuntu 16.04 container
 if [[ "$ROS_DISTRO" == "kinetic" ]] && ! [ "$IN_DOCKER" ]; then
   travis_time_start build_docker_image
-  docker build -t industrial-ci/xenial .ci_config
+  docker build -t industrial-ci/xenial .
   travis_time_end  # build_docker_image
 
   travis_time_start run_travissh_docker
@@ -67,7 +66,6 @@ if [[ "$ROS_DISTRO" == "kinetic" ]] && ! [ "$IN_DOCKER" ]; then
       -e BUILDER \
       -e CATKIN_PARALLEL_JOBS \
       -e CATKIN_PARALLEL_TEST_JOBS \
-      -e CI_PARENT_DIR \
       -e NOT_TEST_BUILD \
       -e NOT_TEST_INSTALL \
       -e PRERELEASE \
@@ -80,7 +78,7 @@ if [[ "$ROS_DISTRO" == "kinetic" ]] && ! [ "$IN_DOCKER" ]; then
       -e ROSWS \
       -e TARGET_PKGS \
       -e USE_DEBROS_DISTRO \
-      -v $(pwd):/root/ci_src industrial-ci/xenial \
+      -v $TARGET_REPO_PATH/:/root/ci_src industrial-ci/xenial \
       /bin/bash -c "cd /root/ci_src; source .ci_config/travis.sh;"
   retval=$?
   if [ $retval -eq 0 ]; then HIT_ENDOFSCRIPT=true; success 0; else exit; fi  # Call  travis_time_end  run_travissh_docker
@@ -90,7 +88,6 @@ travis_time_start init_travis_environment
 # Define more env vars
 BUILDER=catkin
 ROSWS=wstool
-export DOWNSTREAM_REPO_NAME=${PWD##*/}
 if [ ! "$CATKIN_PARALLEL_JOBS" ]; then export CATKIN_PARALLEL_JOBS="-p4"; fi
 if [ ! "$CATKIN_PARALLEL_TEST_JOBS" ]; then export CATKIN_PARALLEL_TEST_JOBS="$CATKIN_PARALLEL_JOBS"; fi
 if [ ! "$ROS_PARALLEL_JOBS" ]; then export ROS_PARALLEL_JOBS="-j8"; fi
@@ -111,13 +108,13 @@ fi
 if [ ! "$UPSTREAM_WORKSPACE" ]; then export UPSTREAM_WORKSPACE="debian"; fi
 
 git branch --all
-if [ "`git diff origin/master FETCH_HEAD $CI_PARENT_DIR`" != "" ] ; then DIFF=`git diff origin/master FETCH_HEAD $CI_PARENT_DIR | grep .*Subproject | sed s'@.*Subproject commit @@' | sed 'N;s/\n/.../'`; (cd $CI_PARENT_DIR/;git log --oneline --graph --left-right --first-parent --decorate $DIFF) | tee /tmp/$$-travis-diff.log; grep -c '<' /tmp/$$-travis-diff.log && exit 1; echo "ok"; fi
+if [ "`git diff origin/master FETCH_HEAD $ICI_PKG_PATH`" != "" ] ; then DIFF=`git diff origin/master FETCH_HEAD $ICI_PKG_PATH | grep .*Subproject | sed s'@.*Subproject commit @@' | sed 'N;s/\n/.../'`; (cd $CI_MAIN_PKG/;git log --oneline --graph --left-right --first-parent --decorate $DIFF) | tee /tmp/$$-travis-diff.log; grep -c '<' /tmp/$$-travis-diff.log && exit 1; echo "ok"; fi
 
 travis_time_end  # init_travis_environment
 
 travis_time_start setup_ros
 
-echo "Testing branch $TRAVIS_BRANCH of $DOWNSTREAM_REPO_NAME"
+echo "Testing branch $TRAVIS_BRANCH of $TARGET_REPO_NAME"  # $TARGET_REPO_NAME is the repo where this job is triggered from, and the variable is expected to be passed externally (industrial_ci/travis.sh should be taking care of it)
 # Set apt repo
 sudo -E sh -c 'echo "deb $ROS_REPOSITORY_PATH `lsb_release -cs` main" > /etc/apt/sources.list.d/ros-latest.list'
 # Common ROS install preparation
@@ -171,8 +168,8 @@ travis_time_start setup_rosws
 
 ## BEGIN: travis' install: # Use this to install any prerequisites or dependencies necessary to run your build ##
 # Create workspace
-mkdir -p ~/ros/ws_$DOWNSTREAM_REPO_NAME/src
-cd ~/ros/ws_$DOWNSTREAM_REPO_NAME/src
+mkdir -p ~/ros/ws_$TARGET_REPO_NAME/src
+cd ~/ros/ws_$TARGET_REPO_NAME/src
 case "$UPSTREAM_WORKSPACE" in
 debian)
     echo "Obtain deb binary for upstream packages."
@@ -180,12 +177,12 @@ debian)
 file) # When UPSTREAM_WORKSPACE is file, the dependended packages that need to be built from source are downloaded based on $ROSINSTALL_FILENAME file.
     $ROSWS init .
     # Prioritize $ROSINSTALL_FILENAME.$ROS_DISTRO if it exists over $ROSINSTALL_FILENAME.
-    if [ -e $CI_SOURCE_PATH/$ROSINSTALL_FILENAME.$ROS_DISTRO ]; then
+    if [ -e $TARGET_REPO_PATH/$ROSINSTALL_FILENAME.$ROS_DISTRO ]; then
         # install (maybe unreleased version) dependencies from source for specific ros version
-        $ROSWS merge file://$CI_SOURCE_PATH/$ROSINSTALL_FILENAME.$ROS_DISTRO
-    elif [ -e $CI_SOURCE_PATH/$ROSINSTALL_FILENAME ]; then
+        $ROSWS merge file://$TARGET_REPO_PATH/$ROSINSTALL_FILENAME.$ROS_DISTRO
+    elif [ -e $TARGET_REPO_PATH/$ROSINSTALL_FILENAME ]; then
         # install (maybe unreleased version) dependencies from source
-        $ROSWS merge file://$CI_SOURCE_PATH/$ROSINSTALL_FILENAME
+        $ROSWS merge file://$TARGET_REPO_PATH/$ROSINSTALL_FILENAME
     fi
     ;;
 http://* | https://*) # When UPSTREAM_WORKSPACE is an http url, use it directly
@@ -197,19 +194,19 @@ esac
 # download upstream packages into workspace
 if [ -e .rosinstall ]; then
     # ensure that the downstream is not in .rosinstall
-    $ROSWS rm $DOWNSTREAM_REPO_NAME || true
+    $ROSWS rm $TARGET_REPO_NAME || true
     $ROSWS update
 fi
-# CI_SOURCE_PATH is the path of the downstream repository that we are testing. Link it to the catkin workspace
-ln -s $CI_SOURCE_PATH .
+# TARGET_REPO_PATH is the path of the downstream repository that we are testing. Link it to the catkin workspace
+ln -s $TARGET_REPO_PATH .
 
 # Disable metapackage
-find -L . -name package.xml -print -exec ${CI_SOURCE_PATH}/$CI_PARENT_DIR/check_metapackage.py {} \; -a -exec bash -c 'touch `dirname ${1}`/CATKIN_IGNORE' funcname {} \;
+find -L . -name package.xml -print -exec ${ICI_PKG_PATH}/check_metapackage.py {} \; -a -exec bash -c 'touch `dirname ${1}`/CATKIN_IGNORE' funcname {} \;
 
 source /opt/ros/$ROS_DISTRO/setup.bash # ROS_PACKAGE_PATH is important for rosdep
 # Save .rosinstall file of this tested downstream repo, only during the runtime on travis CI
 if [ ! -e .rosinstall ]; then
-    echo "- git: {local-name: $DOWNSTREAM_REPO_NAME, uri: 'http://github.com/$TRAVIS_REPO_SLUG'}" >> .rosinstall
+    echo "- git: {local-name: $TARGET_REPO_NAME, uri: 'http://github.com/$TRAVIS_REPO_SLUG'}" >> .rosinstall
 fi
 
 travis_time_end  # setup_rosws
@@ -225,8 +222,8 @@ travis_time_end  # before_script
 travis_time_start rosdep_install
 
 # Run "rosdep install" command. Avoid manifest.xml files if any.
-if [ -e ${CI_SOURCE_PATH}/$CI_PARENT_DIR/rosdep-install.sh ]; then
-    ${CI_SOURCE_PATH}/$CI_PARENT_DIR/rosdep-install.sh
+if [ -e ${ICI_PKG_PATH}/rosdep-install.sh ]; then
+    ${ICI_PKG_PATH}/rosdep-install.sh
 fi
 
 travis_time_end  # rosdep_install
@@ -234,8 +231,8 @@ travis_time_end  # rosdep_install
 # Start prerelease, and once it finishs then finish this script too.
 # This block needs to be here (i.e. After rosdep is done) because catkin_test_results isn't available until up to this point.
 travis_time_start prerelease_from_travis_sh
-if [ "$PRERELEASE" == true ] && [ -e ${CI_SOURCE_PATH}/$CI_PARENT_DIR/ros_pre-release.sh ]; then
-  source ${CI_SOURCE_PATH}/${CI_PARENT_DIR}/ros_pre-release.sh && run_ros_prerelease
+if [ "$PRERELEASE" == true ] && [ -e ${ICI_PKG_PATH}/ros_pre-release.sh ]; then
+  source ${ICI_PKG_PATH}/ros_pre-release.sh && run_ros_prerelease
   retval_prerelease=$?
   if [ $retval_prerelease -eq 0 ]; then HIT_ENDOFSCRIPT=true; success 0; else error; fi  # Internally called travis_time_end for prerelease_from_travis_sh
   # With Prerelease option, we want to stop here without running the rest of the code.
@@ -253,8 +250,8 @@ travis_time_start catkin_build
 ## BEGIN: travis' script: # All commands must exit with code 0 on success. Anything else is considered failure.
 source /opt/ros/$ROS_DISTRO/setup.bash # re-source setup.bash for setting environmet vairable for package installed via rosdep
 # for catkin
-if [ "${_TARGET_PKGS// }" == "" ]; then export _TARGET_PKGS=`catkin_topological_order ${CI_SOURCE_PATH} --only-names`; fi  # `_TARGET_PKGS` (default: not set): If not set, the packages in the output of `catkin_topological_order` from the source space of your repo are to be set. This is also used to fill `PKGS_DOWNSTREAM` if it is not set. 
-if [ "${_PKGS_DOWNSTREAM// }" == "" ]; then export _PKGS_DOWNSTREAM=$( [ "${BUILD_PKGS_WHITELIST// }" == "" ] && echo "$_TARGET_PKGS" || echo "$BUILD_PKGS_WHITELIST"); fi
+if [ "${TARGET_PKGS// }" == "" ]; then export TARGET_PKGS=`catkin_topological_order ${TARGET_REPO_PATH} --only-names`; fi
+if [ "${PKGS_DOWNSTREAM// }" == "" ]; then export PKGS_DOWNSTREAM=$( [ "${BUILD_PKGS_WHITELIST// }" == "" ] && echo "$TARGET_PKGS" || echo "$BUILD_PKGS_WHITELIST"); fi
 if [ "$BUILDER" == catkin ]; then catkin build -i -v --summarize  --no-status $BUILD_PKGS_WHITELIST $CATKIN_PARALLEL_JOBS --make-args $ROS_PARALLEL_JOBS            ; fi
 
 travis_time_end  # catkin_build
@@ -272,7 +269,7 @@ if [ "$NOT_TEST_BUILD" != "true" ]; then
 
     if [ "$BUILDER" == catkin ]; then
         source devel/setup.bash ; rospack profile # force to update ROS_PACKAGE_PATH for rostest
-        catkin run_tests -iv --no-deps --no-status $_PKGS_DOWNSTREAM $CATKIN_PARALLEL_TEST_JOBS --make-args $ROS_PARALLEL_TEST_JOBS --
+        catkin run_tests -iv --no-deps --no-status $PKGS_DOWNSTREAM $CATKIN_PARALLEL_TEST_JOBS --make-args $ROS_PARALLEL_TEST_JOBS --
         catkin_test_results build || error
     fi
 
@@ -298,7 +295,7 @@ if [ "$NOT_TEST_INSTALL" != "true" ]; then
     export EXIT_STATUS=0
     # Test if the unit tests in the packages in the downstream repo pass.
     if [ "$BUILDER" == catkin ]; then
-      for pkg in $_PKGS_DOWNSTREAM; do
+      for pkg in $PKGS_DOWNSTREAM; do
         echo "[$pkg] Started testing..."
         rostest_files=$(find install/share/$pkg -iname '*.test')
         echo "[$pkg] Found $(echo $rostest_files | wc -w) tests."
@@ -326,7 +323,7 @@ PATH=/usr/local/bin:$PATH  # for installed catkin_test_results
 PYTHONPATH=/usr/local/lib/python2.7/dist-packages:$PYTHONPATH
 if [ "${ROS_LOG_DIR// }" == "" ]; then export ROS_LOG_DIR=~/.ros/test_results; fi # http://wiki.ros.org/ROS/EnvironmentVariables#ROS_LOG_DIR
 if [ "$BUILDER" == catkin -a -e $ROS_LOG_DIR ]; then catkin_test_results --verbose --all $ROS_LOG_DIR || error; fi
-if [ "$BUILDER" == catkin -a -e ~/ros/ws_$DOWNSTREAM_REPO_NAME/build/ ]; then catkin_test_results --verbose --all ~/ros/ws_$DOWNSTREAM_REPO_NAME/build/ || error; fi
+if [ "$BUILDER" == catkin -a -e ~/ros/ws_$TARGET_REPO_NAME/build/ ]; then catkin_test_results --verbose --all ~/ros/ws_$TARGET_REPO_NAME/build/ || error; fi
 if [ "$BUILDER" == catkin -a -e ~/.ros/test_results/ ]; then catkin_test_results --verbose --all ~/.ros/test_results/ || error; fi
 
 travis_time_end  # after_script
```